shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -1649,7 +1751,9 @@ - (NSArray*)selectionRectsForRange:(UITextRange*)range {
   NSUInteger end = ((FlutterTextPosition*)range.end).index;
   NSMutableArray* rects = [[NSMutableArray alloc] init];
   for (NSUInteger i = 0; i < [_selectionRects count]; i++) {
-    if (_selectionRects[i].position >= start && _selectionRects[i].position <= end) {
+    if (_selectionRects[i].position >= start &&
+        (_selectionRects[i].position < end ||
+         (start == end && _selectionRects[i].position <= end))) {
```

Comment:
In the old behaviour, you would always get one more extra rect (since it was comparing `<= end`). Now, we are correctly using `< end`. But we still want to return the special rect (zero-width) if `start == end` instead of no rects. So we can use the old comparison `<= end` but only if we have `start == end`.

|Text Range|Old rects returned|New rects returned|
|--|--|--|
|0..0|[0] with width zero|[0] with width zero|
|0..1|[0], [1]|[0]|
|0..2|[0], [1], [2]|[0], [1]|

And so on...

Reply:
Gotcha. Looks like the old behavior is a bug then.